### ISSUE TITLE
unable to find /usr/local/opt/gdal/lib/libgdal.31.dylib after homebrew upgrade

### ISSUE THREAD

**Issue by aisiklar** [eclipse/sumo#12359]
Hello,

This morning I brew-upgraded the formulae and casks on my MacBook-Pro (intel). 

Before (last Friday) there were no problems. I was able to execute run.py file (and a newer version edited by me), which executed sumo (-GUI) without any issues. (my previous installation on sumo and related lib. were in early November I suppose.. I think, after that, sumo and/or its dependencies (like gdal) were  upgraded and my not-so-clever attempt to brew upgrade command made all these upgrade to the newer versions)...

Now, I receive the following problem:
-----"
/usr/local/bin/python3.11 /Users/alptekin/Documents/development/sumo-simulator/sumo-exercises/demo1-complex/traci/run.py
dyld[11156]: Library not loaded: /usr/local/opt/gdal/lib/libgdal.31.dylib
  Referenced from: /Users/alptekin/Documents/development/sumo-simulator/sumo/bin/sumo-gui
  Reason: tried: '/usr/local/opt/gdal/lib/libgdal.31.dylib' (no such file), '/usr/local/lib/libgdal.31.dylib' (no such file), '/usr/lib/libgdal.31.dylib' (no such file), '/usr/local/Cellar/gdal/3.6.1/lib/libgdal.31.dylib' (no such file), '/usr/local/lib/libgdal.31.dylib' (no such file), '/usr/lib/libgdal.31.dylib' (no such file)
 Retrying in 1 seconds
/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/main.py:154: UserWarning: Could not connect to TraCI server using port 49505 (TraCI server already finished). Retrying with different port.
  warnings.warn(("Could not connect to TraCI server using port %s (%s)." +
 Retrying in 1 seconds
dyld[11159]: Library not loaded: /usr/local/opt/gdal/lib/libgdal.31.dylib
  Referenced from: /Users/alptekin/Documents/development/sumo-simulator/sumo/bin/sumo-gui
  Reason: tried: '/usr/local/opt/gdal/lib/libgdal.31.dylib' (no such file), '/usr/local/lib/libgdal.31.dylib' (no such file), '/usr/lib/libgdal.31.dylib' (no such file), '/usr/local/Cellar/gdal/3.6.1/lib/libgdal.31.dylib' (no such file), '/usr/local/lib/libgdal.31.dylib' (no such file), '/usr/lib/libgdal.31.dylib' (no such file)
/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/main.py:154: UserWarning: Could not connect to TraCI server using port 49508 (TraCI server already finished). Retrying with different port.
  warnings.warn(("Could not connect to TraCI server using port %s (%s)." +
"-----

here it says something about port number AND a lib file, which is not found...

After trying different things, which also included returning to version 3.6.0 for gdal, I decided to uninstall almost everything related with sumo and re-install...

Which also did not work...
Still the same error.

I have run out of options. I will appreciate a lot if one can return me with a solution or ideas.

thanks and happy new year to all (apparently it was not a very clever start for me, upgrading before a critical demo tomorrow).
regards
alptekin

note: some peculiar things also occur:
I can execute sumo-gui from terminal by writing sumo-gui but cannot just execute it from "applications" anymore... 
The same for netedit... (I really don't care as long as they run... it is just weird)
But as said, executing run.py doesn't work and gives the above problem.

I am still trying to solve the problem. If in case I succeed before help comes, I will announce it here.

**Comment by aisiklar** [comment]
additional info:
at the specified lib, there are libgdal.32.3.6.1.dylib and libgdal.32.dylib and libgdal.dylib files currently.

Also, I see the following error when trying to execute the runner.py file in the tutorials and do a "command-c" after failure and receiving the error given above:
".......

^CTraceback (most recent call last):
  File "/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/main.py", line 97, in connect
    return connection.Connection(host, port, proc, traceFile, traceGetters, label)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/connection.py", line 82, in __init__
    self._socket.connect((host, port))
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/alptekin/Documents/development/sumo-simulator/sumo/tests/complex/tutorial/traci_tls/runner.py", line 131, in <module>
    traci.start([sumoBinary, "-c", "data/cross.sumocfg",
  File "/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/main.py", line 147, in start
    result = init(sumoPort, numRetries, "localhost", label, sumoProcess, doSwitch, traceFile, traceGetters)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/main.py", line 116, in init
    con = connect(port, numRetries, host, proc, 1, traceFile, traceGetters, label)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/alptekin/Documents/development/sumo-simulator/sumo/tools/traci/main.py", line 105, in connect
    time.sleep(waitBetweenRetries)
KeyboardInterrupt
"

thanks.

**Comment by aisiklar** [comment]
Progress:

I think I found the source of the problem and solved it.
the home directory for sumo is changed (in the last upgrade, I think)... 

I am not sure how and why
but after reinstalling sumo and all other stuff, It gave me a new home directory (which I didn't pay attention until reading all the docs for installation again)...
 When I updated the "export SUMO_HOME = ...." with the given directory, all seems to be ok.

Sorry for making a fuss. I hope these comments make use of others...

in summary,
if you are -for whatever reason- updating /upgrading or re-installing SUMO pay attention to the new home directory and don't forget to change it in your bash/zsh profile...

closing this, If I can...
